cockatrice/src/carddbparser/carddatabaseparser.h

```diff
@@ -0,0 +1,22 @@
+#ifndef CARDDATABASE_PARSER_H
+#define CARDDATABASE_PARSER_H
+
+#include <QString>
+#include <QIODevice>
+
+#include "../carddatabase.h"
+
+class ICardDatabaseParser : public QObject
+{
+public:
+    virtual ~ICardDatabaseParser() {}
+    virtual bool getCanParseFile(const QString &name, QIODevice &device) = 0;
```

Comment:
`= 0;` vs `= delete;` ?

Reply:
I want a pure virtual function in order to make the class abstract (non-instanceable).
As far as I understand, with `= delete` the method is not callable, but the class can still be instanced.

Reply:
https://stackoverflow.com/questions/24958270/why-cant-i-declare-a-pure-virtual-function-with-delete Yeah you're right. Didn't realize we could create a fully abstract class in C++ :D

Reply:
Yep, in any other language i'd use an `interface`. In C++ you need some trickery to achieve the same result.